src/enclave/httpsig.h

```diff
@@ -13,297 +13,335 @@
 
 namespace enclave
 {
-  // All HTTP headers are expected to be lowercase
-  static constexpr auto HTTP_HEADER_AUTHORIZATION = "authorization";
-  static constexpr auto HTTP_HEADER_DIGEST = "digest";
-
-  static constexpr auto DIGEST_SHA256 = "SHA-256";
-
-  static constexpr auto AUTH_SCHEME = "Signature";
-  static constexpr auto SIGN_PARAMS_KEYID = "keyId";
-  static constexpr auto SIGN_PARAMS_SIGNATURE = "signature";
-  static constexpr auto SIGN_PARAMS_ALGORITHM = "algorithm";
-  static constexpr auto SIGN_PARAMS_HEADERS = "headers";
-  static constexpr auto SIGN_ALGORITHM = "ecdsa-sha256";
-
-  static constexpr auto SIGN_PARAMS_DELIMITER = ",";
-  static constexpr auto SIGN_PARAMS_HEADERS_DELIMITER = " ";
-
-  std::optional<std::vector<uint8_t>> construct_raw_signed_string(
-    const http::HeaderMap& headers,
-    const std::vector<std::string_view>& headers_to_sign)
+  namespace http
   {
-    std::string signed_string = {};
-
-    bool first = true;
-    bool has_digest = false;
-
-    for (const auto f : headers_to_sign)
+    // All HTTP headers are expected to be lowercase
+    static constexpr auto HTTP_HEADER_AUTHORIZATION = "authorization";
+    static constexpr auto HTTP_HEADER_DIGEST = "digest";
+
+    static constexpr auto DIGEST_SHA256 = "SHA-256";
+
+    static constexpr auto AUTH_SCHEME = "Signature";
+    static constexpr auto SIGN_PARAMS_KEYID = "keyId";
+    static constexpr auto SIGN_PARAMS_SIGNATURE = "signature";
+    static constexpr auto SIGN_PARAMS_ALGORITHM = "algorithm";
+    static constexpr auto SIGN_PARAMS_HEADERS = "headers";
+    static constexpr auto SIGN_ALGORITHM = "ecdsa-sha256";
+
+    static constexpr auto SIGN_HEADER_REQUEST_TARGET = "(request-target)";
+
+    static constexpr auto SIGN_PARAMS_DELIMITER = ",";
+    static constexpr auto SIGN_PARAMS_HEADERS_DELIMITER = " ";
+
+    std::optional<std::vector<uint8_t>> construct_raw_signed_string(
+      std::string verb,
+      const std::string& path,
+      const std::string& query,
+      const http::HeaderMap& headers,
+      const std::vector<std::string_view>& headers_to_sign)
     {
-      const auto h = headers.find(f);
-      if (h == headers.end())
-      {
-        LOG_FAIL_FMT("Signed header {} does not exist", f);
-        return {};
-      }
-
-      // Digest field should be signed.
-      if (f == HTTP_HEADER_DIGEST)
-      {
-        has_digest = true;
-      }
+      std::string signed_string = {};
+      std::string value = {};
+      bool has_digest = false;
+      bool first = true;
 
-      if (!first)
+      for (const auto f : headers_to_sign)
```

Comment:
Changes in this file are here. The rest is indentation due to the `http` namespace.